### ISSUE TITLE
why only focus document can be used save actions? version 0.7

### ISSUE THREAD

**Issue by myoss** [dubreuia/intellij-plugin-save-actions#30]
before upgrade to version 0.7, i used version 0.4, the can auto format all modified file (before do make/debug/other things), today i upgrade to version 0.7 and debug the source code, now why only focus document can be used save actions? when do make/debug can not used

    @Override
    public void beforeDocumentSaving(@NotNull Document document) {
        if (!SaveAllAction.TRIGGERED && isDocumentActive(document)) {
            LOGGER.debug("Document " + document + " is still active, do not execute");
            return;
        }
        for (Project project : ProjectManager.getInstance().getOpenProjects()) {
            PsiFile psiFile = PsiDocumentManager.getInstance(project).getPsiFile(document);
            if (isPsiFileEligible(project, psiFile)) {
                processPsiFile(project, psiFile);
            }
        }
    }

----------------Environment Version----------------
save actions plugin version 0.7
intellij idea version 15.0.3

**Comment by dubreuia** [comment]
Thanks for the issue. I'll check out if I can support both (1) reformat only the focus document (2) reformat all modified documents.

*tl;dr:* It's a side effect of #13

*Long explanation:* to fix #1, I need to make sure to reformat only the file that is opened in the current intellij (there can be multiple instances), but I can't check on file path since it breaks #13, so I'm using the current file focus to see if I need to reformat or not. It might work better if I use the current project focus instead.

**Comment by dubreuia** [comment]
I might just add an option "Reformat only focus document" so I'm marking that as an enhancement since 0.7 is working as I want it to work.

**Comment by myoss** [comment]
I'm looking forward to it.

**Comment by sbmpost** [comment]
@dubreuia: hi there, you may want to take a look at some code I posted here: http://stackoverflow.com/a/35727193
It solves some problems (locked files, not being able to undo, and formatting multiple files).

**Comment by dubreuia** [comment]
I like the make project / compile file options, I might steal your idea. Not sure I can use the multiple files handling part but I'll try. I'll certainly use the part on the undo and the locked files. Thanks a lot I'll keep you posted!

**Comment by sbmpost** [comment]
You are welcome to steal ;-), thanks for working on this

**Comment by dannygod** [comment]
It seems not working with IdeaVim plugin when typing **:w** to save a file due to this issue. Roll back to version 0.4 and everything work fine. Looking forward to this enhancement in future version.

**Comment by dubreuia** [comment]
Corrected that behavior in 0.8. It should work as in 0.4, but works with multiple open instances too
